SortApp/SortApp.BL/Repository/DataRepository.cs

```diff
@@ -0,0 +1,396 @@
+		/// <param name="row">
+		/// Original row as DataRow.
+		/// </param>
+		/// <param name="iterationRows">
+		/// Iteration rows.
+		/// </param>
+		/// <returns>
+		/// Data object created from DataRow.
+		/// </returns>
+		private Data CreateDataFromDataRow(DataRow row, DataRowCollection iterationRows)
+		{
+			Data data = new Data
+			{
+				Id = (int)row["Id"],
+				OriginalData = (string)row["OriginalData"],
+				SortedData = row["SortedData"] != DBNull.Value ? (string)row["SortedData"] : null,
+				Name = row["Name"] != DBNull.Value ? (string)row["Name"] : null,
+				Iterations = new List<Iteration>(iterationRows.Count)
+			};
+
+			foreach (DataRow iterationRow in iterationRows)
+			{
+				data.Iterations.Add(new Iteration(
+					(int)iterationRow["IndexOfFirstMovedElement"],
+					(int)iterationRow["IndexOfSecondMovedElement"],
+					(int)iterationRow["Number"])
+				{
+					ArrayState = (string)iterationRow["ArrayState"]
+				});
+			}
+
+			return data;
+		}
+
+		/// <summary>
+		/// Creates dataAdapter for Iteration.
+		/// </summary>
+		/// <owner>Oleh Petrenko</owner>
+		/// <returns>
+		/// Data adapter for Iteration.
+		/// </returns>
+		private SqlDataAdapter CreateIterationAdapter()
+		{
+			SqlDataAdapter adapter = new SqlDataAdapter();
+
+			//
+			// SelectCommand.
+			//
+			SqlCommand cmdSelect = new SqlCommand("SELECT * FROM Iteration WHERE DataId = @DataId", this.connection);
+			adapter.SelectCommand = cmdSelect;
+
+			SqlParameter parameter = adapter.SelectCommand.Parameters.Add("@DataId", SqlDbType.Int);
+			parameter.SourceColumn = "DataId";
+
+			//
+			// InsertCommand.
+			//
+			SqlCommand cmdInsert = new SqlCommand("INSERT INTO Iteration(ArrayState, IndexOfFirstMovedElement, IndexOfSecondMovedElement, Number, DataId) " +
+												"VALUES(@ArrayState, @IndexOfFirstMovedElement, @IndexOfSecondMovedElement, @Number, @DataId)", this.connection);
+			adapter.InsertCommand = cmdInsert;
+
+			parameter = adapter.InsertCommand.Parameters.Add("@ArrayState", SqlDbType.Text);
+			parameter.SourceColumn = "ArrayState";
+
+			parameter = adapter.InsertCommand.Parameters.Add("@IndexOfFirstMovedElement", SqlDbType.Int);
+			parameter.SourceColumn = "IndexOfFirstMovedElement";
+
+			parameter = adapter.InsertCommand.Parameters.Add("@IndexOfSecondMovedElement", SqlDbType.Int);
+			parameter.SourceColumn = "IndexOfSecondMovedElement";
+
+			parameter = adapter.InsertCommand.Parameters.Add("@Number", SqlDbType.Int);
+			parameter.SourceColumn = "Number";
+
+			parameter = adapter.InsertCommand.Parameters.Add("@DataId", SqlDbType.Int);
+			parameter.SourceColumn = "DataId";
+
+			return adapter;
+		}
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="DataRepository"/> class.
+		/// </summary>
+		/// <owner>Oleh Petrenko</owner>
+		/// <param name="connection">
+		/// SqlConnection entity for connecting to database.
+		/// </param>
+		public DataRepository(SqlConnection connection)
```

Comment:
I think that connection should be created inside this class. It's better to get connection string in constructor.